### ISSUE TITLE
feature suggestion: use input parameter to select texlive version instead of relying on different tags

### ISSUE THREAD

**Issue by zydou** [xu-cheng/latex-action#129]
## Motivation

For a specific reason, I must ensure the seamless compilation of my LaTeX project across multiple texlive versions.

The action `xu-cheng/latex-action` provides support for employing different tags to select various texlive versions. For instance:

```yaml
- uses: xu-cheng/latex-action@v2             # TeXLive 2023
- uses: xu-cheng/latex-action@v2-texlive2022 # TeXLive 2022
- uses: xu-cheng/latex-action@v2-texlive2021 # TeXLive 2021
- uses: xu-cheng/latex-action@v2-texlive2020 # TeXLive 2020
```

An initial approach could involve creating a workflow as follows:

```yaml
jobs:
  build_latex:
    strategy:
      fail-fast: false
      matrix:
        tag: [v2, v2-texlive2022, v2-texlive2021, v2-texlive2020]
    name: texlive-${{ matrix.tag }}
    runs-on: ubuntu-latest
    steps:
      - name: Set up Git repository
        uses: actions/checkout@v3
      - name: Compile LaTeX document
        uses: xu-cheng/latex-action@${{ matrix.tag }}
        with:
          root_file: main.tex
          # other options...
```

However, this approach encounters issues due to the fact that, during the initial step of a GitHub Actions workflow run, all action repositories specified in `uses` fields are fetched **literally**, without parsing any `${{...}}` variables.

A temporary solution is using a `if` condition to choose the appropriate action:

```yaml
      - name: Compile LaTeX document
        if: ${{ matrix.tag == 'v2-texlive2022' }}
        uses: xu-cheng/latex-action@v2-texlive2022
        with:
          root_file: main.tex

      - name: Compile LaTeX document
        if: ${{ matrix.tag == 'v2-texlive2021' }}
        uses: xu-cheng/latex-action@v2-texlive2021
        with:
          root_file: main.tex

      # copy above block multiple times
```

It would be nice if we can add an input parameter to select the texlive version directly.

## Solution

TL;DR: use `composite` action instead of `docker` action. checkout this [action.yml](https://github.com/zydou/latex-action/blob/master/action.yml)

```yaml
runs:
  using: composite
  steps:
    - shell: bash
      run: >
        docker run --rm
        -v "${PWD}":/github/workspace
        -v "${GITHUB_ACTION_PATH}/entrypoint.sh":/entrypoint.sh
        -w /github/workspace
        --entrypoint "/entrypoint.sh"
        ghcr.io/xu-cheng/texlive-full:${{ inputs.texlive_version }}
        "${{ inputs.root_file }}"
        "${{ inputs.glob_root_file }}"
        # other inputs...
```

I maintained a fork that use `Debian` based texlive image at [zydou/latex-action](https://github.com/zydou/latex-action). It has two additional input parameters `texlive_version` and `debian_release` to select the texlive version and Debian release, respectively.

```yaml
      - name: Compile LaTeX document
        uses: zydou/latex-action@v3
        with:
          root_file: main.tex
          texlive_version: 2022
          debian_release: bookworm
```

What do you think of this? Please feel free to close this issue if you think it is trivial, since using `if` condition is indeed a feasible solution.

If you think this idea is favorable, I can send a PR to add this feature. However, it appears that utilizing both an input parameter and tags might be redundant. The back compatibility needs to be discussed. In my own fork, I just bumped the major version number to indicate the breaking change.

**Comment by xu-cheng** [comment]
Thanks for the advice. This is indeed a good idea. However, it may has the a minor issue that user defined env may not be passed to the container. Currently, GitHub action will automatically pass any defined environment variables inside the workflow to the container. If we want to use this approach, we would have to define a list of common envs to be passed to the container.

Nevertheless, a PR is welcome.

**Comment by zydou** [comment]
I am not familiar with the user-defined environment variables that affect texlive. Based on the information provided [here](https://github.com/xu-cheng/latex-docker/blob/0cacb0b783a9ab566a99f78d1de527a2a0a11dd2/texlive.profile#L3-L9), I believe these environment variables should not be modified:

```
TEXDIR opt/texlive/texdir
TEXMFLOCAL opt/texlive/texmf-local
TEXMFSYSVAR opt/texlive/texdir/texmf-var
TEXMFSYSCONFIG opt/texlive/texdir/texmf-config
TEXMFVAR root/texlive/texmf-var
TEXMFCONFIG root
```

From the README, it appears that `$TEXINPUTS` is an environment variable we should pay attention to. We can easily pass this variable to `entrypoint.sh`. However, I am unsure if there are any other environment variables we need to consider.

**Comment by xu-cheng** [comment]
Closed by #130
